modules/gateways/veritrans.php

```diff
@@ -360,10 +360,24 @@ function veritrans_link($params)
         $snapToken = $snap_transaction->token;
         $redirect_url = $snap_transaction->redirect_url;
         // error_log(" ############# TOKEN ::: ".$snapToken);
+
+        // save snap redirect url to invoice notes
+        $command = 'UpdateInvoice';
+        $postData = array(
+            'invoiceid' => $invoiceId,
+            'notes' => $redirect_url
+        );
+        $results = localAPI($command, $postData);
+
     } catch (Exception $e) {
         // error_log('Caught exception: ',  $e->getMessage(), "\n");
-        if (preg_match('/utilized/',$e->getMessage()) ){
-            return "<h4> Awaiting Your Payment </h4>";
+        if (preg_match('/utilized|digunakan/',$e->getMessage()) ){
+            $command = 'GetInvoice';
+            $postData = array(
+                'invoiceid' => $invoiceId
+            );
```

Comment:
why is the real localAPI function never get called? you are just assigning value to variable without calling any meaningful function.